tests/e2e/core-tests/specs/shopper/front-end-single-product.test.js

```diff
@@ -67,6 +69,50 @@ const runSingleProductPageTest = () => {
 			await expect(page).toMatchElement('.cart-empty', {text: 'Your cart is currently empty.'});
 		});
 	});
+
+	describe('Grouped Product Page', () => {
+		beforeAll(async () => {
+			await merchant.login();
+			groupedPostIdValue = await createGroupedProduct();
+			await merchant.logout();
+		})
+
+		it('should be able to add grouped products to the cart', async () => {
+			// Add a grouped product to cart
+			await shopper.goToProduct(groupedPostIdValue);
+			await page.waitForSelector('form.grouped_form');
+			await shopper.addToCart();
+			await expect(page).toMatchElement('.woocommerce-error',
+			 {text: 'Please choose the quantity of items you wish to add to your cart…'});
+			const quantityFields = await page.$$('div.quantity input.qty');
+			await quantityFields[0].click({clickCount: 3});
+			await quantityFields[0].type('5');
+			await quantityFields[1].click({clickCount: 3});
+			await quantityFields[1].type('5');
+			await shopper.addToCart();
+			await expect(page).toMatchElement('.woocommerce-message',
+			 {text: '“'+simpleProductName+' 1” and “'+simpleProductName+' 2” have been added to your cart.'});
+
+			// Verify cart contents
+			await shopper.goToCart();
+			await shopper.productIsInCart(simpleProductName+' 1');
+			await shopper.productIsInCart(simpleProductName+' 2');
+
+			// Remove items from cart
+			await shopper.removeFromCart(simpleProductName+' 1');
```

Comment:
Since this particular portion of the test is focused on being able to add the items to the cart (`it('should be able to add grouped products to the cart') {}`), I think we can break out the removal of items from the cart piece into a separate test, something like `it('should be able to remove items from cart') {}`. Then we can do a similar check as on lines 97-99 to verify that the products are not in the cart/the cart is empty.

Reply:
Well, that's right. I followed examples from above inside the test. I did it for all three scenarios now.

Reply:
Thanks!